language-support/scala/codegen/src/main/scala/com/digitalasset/lf/codegen/CodeGen.scala

```diff
@@ -220,10 +192,10 @@ object CodeGen {
     filePlans ++ specialPlans
   }
 
-  private[this] def splitNTDs[RT, VT](definitions: List[ScopedDataType.DT[RT, VT]]): (
-      List[ScopedDataType[Record[RT]]],
-      List[ScopedDataType[Variant[VT]]],
-      List[ScopedDataType[Enum]],
+  private[this] def splitNTDs[RT, VT](definitions: Vector[ScopedDataType.DT[RT, VT]]): (
+      Vector[ScopedDataType[Record[RT]]],
+      Vector[ScopedDataType[Variant[VT]]],
+      Vector[ScopedDataType[Enum]],
```

Comment:
Re: https://github.com/digital-asset/daml/issues/13764

Switching from `List` to `Vector` was supposed to be even fast as we started from a `Vector` and this avoided copying into a `List`, but I may have overlooked something.